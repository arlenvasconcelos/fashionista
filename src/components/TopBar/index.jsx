import Topbar from './TopBar'

export default Topbar