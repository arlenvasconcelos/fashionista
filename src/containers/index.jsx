import App from './App/App';
import Products from './Products/Products';
import Cart from './Cart/Cart';
import SingleProduct from './SingleProduct/SingleProduct';

export {App, Products, Cart, SingleProduct}