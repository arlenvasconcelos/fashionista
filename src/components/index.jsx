import CartDetails from './CartDetails/CartDetails';
import CartFooter from './CartFooter/CartFooter';
import CartHeader from './CartHeader/CartHeader';
import Drawer from './Drawer/Drawer';
import Loading from './Loading/Loading';
import NavSearch from './NavSearch/NavSearch';
import NoImage from './NoImage/NoImage';
import Notification from './Notification/Notification';
import Product from './Product/Product';
import ProductDetails from './ProductDetails/ProductDetails';
import ProductImage from './ProductImage/ProductImage';
import ProductPrice from './ProductPrice/ProductPrice';
import TopBar from './TopBar/TopBar';

export {
  CartDetails, CartFooter, CartHeader, Drawer, Loading,
  NavSearch, NoImage, Notification, Product, ProductDetails, 
  ProductImage, ProductPrice, TopBar,
}