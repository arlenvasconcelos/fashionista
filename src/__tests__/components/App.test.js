import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import { Provider } from "react-redux";
import Adapter from "enzyme-adapter-react-16";
import createStore from "redux-mock-store";

import { Products, App, Cart } from "../../containers";
import { Loading } from "../../components";

Enzyme.configure({ adapter: new Adapter() });

describe("App Test", () => {
  const mockStore = createStore();
  const INITIAL_STATE = {
    products: {
      allProducts: [],
      filteredProducts: [],
      openedFilter: false,
      errorStatus: false,
      errorMessage: "Error message",
    },
    cart: {
      items: [],
      open: false,
    },
  };
  const store = mockStore(INITIAL_STATE);

  const wrapper = shallow(
    <Provider store={store}>
      <App />
    </Provider>
  );

  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should find Loading Component", () => {
    expect(wrapper.find(Products)).toBeTruthy();
  });

  it("should find Loading Component", () => {
    expect(wrapper.find(Loading)).toBeTruthy();
  });

  it("should find Cart Component", () => {
    expect(wrapper.find(Cart)).toBeTruthy();
  });
});
