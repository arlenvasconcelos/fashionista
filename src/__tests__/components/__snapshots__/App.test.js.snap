// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test should match the snapshot 1`] = `"<div class=\\"app\\"><header class=\\"topbar\\"><div class=\\"container\\"><div class=\\"topbar__content\\"><a href=\\"/\\"><div class=\\"topbar__logo\\">FASHIONISTA</div></a><div class=\\"topbar__icons\\"><div class=\\"navsearch--ontopbar \\"><div class=\\"navsearch__icon\\"><button><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><circle cx=\\"11\\" cy=\\"11\\" r=\\"8\\"></circle><line x1=\\"21\\" y1=\\"21\\" x2=\\"16.65\\" y2=\\"16.65\\"></line></svg></button></div></div><button class=\\"icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><circle cx=\\"9\\" cy=\\"21\\" r=\\"1\\"></circle><circle cx=\\"20\\" cy=\\"21\\" r=\\"1\\"></circle><path d=\\"M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6\\"></path></svg></button></div></div></div></header><div class=\\"drawer inactive\\"><div class=\\"cart\\"><div class=\\"cart__header\\"><button><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><line x1=\\"5\\" y1=\\"12\\" x2=\\"19\\" y2=\\"12\\"></line><polyline points=\\"12 5 19 12 12 19\\"></polyline></svg></button><span>Carrinho (0)</span></div><div class=\\"cart__details\\"></div><div class=\\"cart__footer\\"><div class=\\"footer__content\\"><p class=\\"content__row\\">Total: <span>0.00</span></p><p class=\\"content__row\\">Desconto: <span>0.00</span></p><p class=\\"content__row--total\\">Total com desconto: <span>0.00</span></p></div><button>CONCLUIR COMPRA</button></div></div></div><div class=\\"loading\\" data-testid=\\"loading\\"></div><footer class=\\"footer\\" style=\\"position:fixed\\"><div class=\\"container\\"><h3>Fashionista E-commerce</h3><p>Acompanhe-nos tamb√©m nas redes sociais</p><div class=\\"footer__icons\\"><a href=\\"#\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><rect x=\\"2\\" y=\\"2\\" width=\\"20\\" height=\\"20\\" rx=\\"5\\" ry=\\"5\\"></rect><path d=\\"M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z\\"></path><line x1=\\"17.5\\" y1=\\"6.5\\" x2=\\"17.51\\" y2=\\"6.5\\"></line></svg></a><a href=\\"#\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><path d=\\"M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z\\"></path></svg></a><a href=\\"#\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><path d=\\"M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z\\"></path></svg></a></div></div></footer></div>"`;
